config/routes.rb

```diff
@@ -300,14 +300,14 @@
   # city_slug: appleton and slug: sample-commitchange-nonprofit, then `slugged_nonprofit_path(nonprofit)` would return:
   # /wi/appleton/sample-commitchange-nonprofit
   direct(:slugged_nonprofit) do |model, options|
-    { controller: "nonprofits", action: "show", state_code: model.state_code_slug, city: model.city_slug, name: model.slug }.merge(options)
+    nonprofit_location_path(state_code: model.state_code_slug, city: model.city_slug, name: model.slug, **options)
```

Comment:
For some reason, when using `slugged_nonprofit`, the previous routes generated differed based on what page the url was generated on. Why? I don't know. This change seems more reliable.